core/modules/hubotModule.js

```diff
@@ -0,0 +1,133 @@
+    descriptor = 'hubot.json',
+    pkg = 'package.json',
+    Robot = require.once('./hubot/robot.js');
+
+var verifyModuleDescriptior = function (hj, disabled) {
+    if (!hj.name || !hj.startup || !hj.version) {
+        return false;
+    }
+    
+    if (disabled === true && exports.disabledConfig 
+        && exports.disabledConfig[hj.name] && exports.disabledConfig[hj.name] === true) {
+        return false;
+    }
+    return true;
+};
+
+exports.verifyModule = function (location, disabled) {
+    var stat = fs.statSync(location);
+    if (!stat.isDirectory()) {
+        return null;
+    }
+    
+    var folderPath = path.resolve(location),
+        desc = path.join(folderPath, './' + descriptor),
+        pack = path.join(folderPath, './' + pkg),
+        hj;
+
+    try {
+        fs.statSync(desc);
+        hj = require.once(desc);
+    }
+    catch (e) {
+        var l;
+        try {
+            fs.statSync(pack);
+            var p = require(pack);
+            hj = Robot.generateHubotJson(folderPath, p.main);
+            hj.name = p.name;
+            hj.version = p.version;
+        }
+        catch (e) {
+            var files = fs.readdirSync(folderPath);
+            if (files.length !== 1) {
+                return null;
+            }
+            hj = Robot.generateHubotJson(folderPath, files[0]);
+        }
+        
+        fs.writeFileSync(desc, JSON.stringify(hj, null, 4), 'utf8');
+    }
+
+    if (!verifyModuleDescriptior(hj, disabled)) {
+        return null;
+    }
+    
+    if (!hj.folderPath) {
+        hj.folderPath = folderPath;
+    }
+    return hj;
+};
+
+exports.listModules = function (disabled) {
+    var data = files.filesInDirectory('./' + modulesDir),
+        modules = {};
+    
+    for (var i = 0; i < data.length; i++) {
+        try {
+            var candidate = path.resolve(path.join(modulesDir, data[i])),
+                output = exports.verifyModule(candidate, disabled);
+            if (output) {
+                modules[output.name] = output;
+            }
+            else {
+                console.debug('Skipping "' + data[i] + '". It isn\'t a Hubot module.');
+            }
+        } catch (e) {
+            console.debug('A failure occured while listing "' + data[i] + '". It doesn\'t appear to be a module.');
+            console.critical(e);
+            continue;
+        }
+    }
+    return modules;
+};
+
+var createHelp = function (module) {
+    return function (commandPrefix) {
+        var h = [];
+        for (var i = 0; i < module.help.length; i++) {
+            var l = [];
+            for (var j = 0; j < module.help[i].length; j++) {
+                l.push(module.help[i][j].replace(/{{commandPrefix}}/g, commandPrefix));
+            }
+            h.push(l);
+        }
+        return h;
+    };
+};
+
+exports.loadModule = function (module) {
+    if (!global.coffeescriptLoaded && module.startup.endsWith('.coffee')) {
+        require("coffee-script").register();
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530854170/issues/source?bid=3282987&fileBranchId=3283570#l117)